atariari/methods/encoders.py

```diff
@@ -124,6 +136,10 @@ def __init__(self, input_channels, args):
             )
         self.train()
 
+    @property
+    def convolution_depth(self):
+        return self.main[4].out_channels
```

Comment:
The name `convolution_depth` might be a bit vague, since it doesn't tell what is the layer we are referring to. Might be better to call the property `depths` and then call `self.encoder.depths[-2]`
```
def depths(self):
    return self._depths
```

Reply:
From a theoretical standpoint what's the benefit of using the 2nd to last depth as the input for the Local-Local and Global-Local score functions? In our experiments we are using the final layer for ImpalaCNN.

Reply:
We want to ensure the receptive field for each patch in the feature map is limited (using the second last layer in NatureCNN limits the receptive field to 1/16th of the full observation). This means that loss is distributed across all the regions of the input, and the encoder can't resort to using only one predictable feature. 

You might want to calculate the receptive field for different layers in the Impala architecture and use the appropriate layer.

Reply:
This depths[-2] is something we may want to vary depending on our encoder architecture. What do you think about renaming this to something like `intermediate_layer_depth`?

Reply:
I see, yes using a fixed layer would be problematic. I think your original solution makes sense then, how about renaming it to `local_layer_depth`?

Reply:
I like that name. I made the change!